compiler/rustc_const_eval/src/interpret/eval_context.rs

```diff
@@ -291,13 +299,6 @@ impl<'tcx> fmt::Display for FrameInfo<'tcx> {
     }
 }
 
-impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> HasDataLayout for InterpCx<'mir, 'tcx, M> {
-    #[inline]
-    fn data_layout(&self) -> &TargetDataLayout {
-        &self.tcx.data_layout
-    }
-}
```

Comment:
Why did you move this? It was deliberately close to the other `Has*` impls for `InterpCx`

Reply:
In the previous PR I had to move `ConstEvalErr` (and some other structs that depend on it) to `rustc_middle`. I pasted this back in the wrong place here, sry. I'll fix this to what it was before.